### PR TITLE
fix(TS): EventSpec recognition

### PR DESCRIPTION
**Pull request by ShaMan123**
## Description

This PR fixes the recognition of the EventSpec generic type that was lost by Group.
Emphasizes the for #8394 
Disclaimer: TS has taken a toll from me in this inane effort. Nothing good to say about it today.

### CODE REVIEW

File: src/static_canvas.class.ts [hunk 1]
```diff
@@ -257,6 +261,28 @@ export class StaticCanvas extends createCollectionMixin(
...
+
+  // TODO: move to canvas
+  interactive: boolean;
+  upperCanvasEl: HTMLCanvasElement;
+  contextTop: CanvasRenderingContext2D;
+  wrapperEl: HTMLDivElement;
+  cacheCanvasEl: HTMLCanvasElement;
+  protected _isCurrentlyDrawing: boolean;
+  freeDrawingBrush: BaseBrush;
+  _activeObject: FabricObject;
```
Comment (ShaMan123) [inline src/static_canvas.class.ts:275]: All of there belong to canvas along with `drawControls` and `_applyCanvasStyle`

**Review by ShaMan123 — COMMENTED** [review]
I think we should consider making `FabricObjectSVGExportMixin` a mixer function like collection and drop `applyMixins`

I need this merged

**Comment by asturur** [comment]
Well i think there is a balance between everyhing magically typed and type safety.
Right now you lean too much on everything typed and that is tiring

File: src/static_canvas.class.ts (hunk 1 above)
Comment (asturur) [reply src/static_canvas.class.ts:275]: So how this passed before withtout the ts-nocheck on top?
How did i get a clea build without those types?

**Review by asturur — APPROVED** [review]

**Comment by ShaMan123** [comment]
It was because of the return type of collection mixin that was any

**Pull request merged**